crates/bitwarden-crypto/src/service/mod.rs

```diff
@@ -0,0 +1,197 @@
+use std::sync::{Arc, RwLock};
+
+use crate::{AsymmetricKeyRef, Decryptable, Encryptable, KeyRef, SymmetricKeyRef, UsesKey};
+
+mod context;
+
+mod key_store;
+
+use context::ReadWriteGlobal;
+pub use context::{CryptoServiceContext, ReadOnlyGlobal};
+pub use key_store::create_key_store;
+use key_store::KeyStore;
+
+#[derive(Clone)]
+pub struct CryptoService<SymmKeyRef: SymmetricKeyRef, AsymmKeyRef: AsymmetricKeyRef> {
+    // We use an Arc<> to make it easier to pass this service around, as we can
+    // clone it instead of passing references
+    key_stores: Arc<RwLock<Keys<SymmKeyRef, AsymmKeyRef>>>,
```

Comment:
**question**: This seems to be one instance of Keys, is there a reason to use a plural name here?

Reply:
I think I've refactored this a couple of times and this was from a time where things were structured differently, I can change it.